### PR TITLE
Add required python packages

### PR DESCRIPTION
**Pull request by Gorcenski**
I've updated the Dockerfile to include the absolute minimal set of packages necessary to build basic models for the TWDE Datalab exercise. There is some trickery involved in what versions get installed.

- First, upgrade pip with `pip3 install --upgrade pip`. This should bring us from pip 9.0.x to pip 18.x
- Next install the required packages with a specific order:
```sh
pip install numpy==1.14.3
pip install python-dateutil
pip install --no-deps pandas==0.23.0
pip install dvc[gs]
pip install scipy==1.1.0
```
- Note that it is necessary to install `dvc[gs]` to include the required packages for using Google Storage
- Finally, I had to `apk add` the following: `openblas-dev` and `gfortran`. These were necessary to get scipy to build.

This package coverage should handle the 90% case of what we might want to use in the datalab. I put these requirements in a separate requirements file, and for adding other packages that we might need, we should just be able to add another requirements file. This way, we should minimize any wonky version conflicts.

Building the Docker image takes a while, but it completed successfully on my machine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 
-RUN apk --no-cache add python3 openssl ca-certificates
+COPY requirements.txt .
+COPY requirements requirements/
+
+RUN apk --no-cache add python3 openssl ca-certificates openblas-dev gfortran
 RUN apk --update add --virtual build-dependencies python3-dev libffi-dev build-base linux-headers \
-  && pip3 install --no-cache-dir dvc \
+  && pip3 install --no-cache-dir --upgrade pip \
+  && pip install --no-cache-dir -r requirements.txt \
```
Comment (dtsato) [inline Dockerfile:9]: Shouldn't this be `pip3 install`? I only had the symlink from pip3 to pip after that.. or after upgrading then pip in the PATH is already pip3?

**Review by dtsato — APPROVED** [review]
Apart from that question, LGTM!

File: Dockerfile (hunk 1 above)
Comment (Gorcenski) [reply Dockerfile:9]: `pip3 install --upgrade pip` makes `pip` point to the python3 version by default, which is the desired behavior

Comment (dtsato) [reply Dockerfile:9]: Cool, in that case, we don't need the `ln` on line 11

**Pull request merged**